include/dav1d/headers.rs

```diff
@@ -109,22 +109,44 @@ impl TryFrom<Dav1dTxfmMode> for Rav1dTxfmMode {
 }
 
 pub type Dav1dFilterMode = u8;
-pub const DAV1D_FILTER_SWITCHABLE: Dav1dFilterMode = 4;
-pub const DAV1D_N_FILTERS: usize = 4;
-pub const DAV1D_FILTER_BILINEAR: Dav1dFilterMode = 3;
-pub const DAV1D_N_SWITCHABLE_FILTERS: usize = 3;
-pub const DAV1D_FILTER_8TAP_SHARP: Dav1dFilterMode = 2;
-pub const DAV1D_FILTER_8TAP_SMOOTH: Dav1dFilterMode = 1;
-pub const DAV1D_FILTER_8TAP_REGULAR: Dav1dFilterMode = 0;
-
-pub type Rav1dFilterMode = u8;
-pub const RAV1D_FILTER_SWITCHABLE: Rav1dFilterMode = DAV1D_FILTER_SWITCHABLE;
-pub const RAV1D_N_FILTERS: usize = DAV1D_N_FILTERS;
-pub const RAV1D_FILTER_BILINEAR: Rav1dFilterMode = DAV1D_FILTER_BILINEAR;
-pub const RAV1D_N_SWITCHABLE_FILTERS: usize = DAV1D_N_SWITCHABLE_FILTERS;
-pub const RAV1D_FILTER_8TAP_SHARP: Rav1dFilterMode = DAV1D_FILTER_8TAP_SHARP;
-pub const RAV1D_FILTER_8TAP_SMOOTH: Rav1dFilterMode = DAV1D_FILTER_8TAP_SMOOTH;
-pub const RAV1D_FILTER_8TAP_REGULAR: Rav1dFilterMode = DAV1D_FILTER_8TAP_REGULAR;
+pub const DAV1D_N_SWITCHABLE_FILTERS: usize = Rav1dFilterMode::N_SWITCHABLE_FILTERS as usize;
+pub const DAV1D_N_FILTERS: usize = Rav1dFilterMode::N_FILTERS as usize;
+pub const DAV1D_FILTER_SWITCHABLE: Dav1dFilterMode = Rav1dFilterMode::Switchable as Dav1dFilterMode;
+pub const DAV1D_FILTER_BILINEAR: Dav1dFilterMode = Rav1dFilterMode::Bilinear as Dav1dFilterMode;
+pub const DAV1D_FILTER_8TAP_SHARP: Dav1dFilterMode = Rav1dFilterMode::Sharp8Tap as Dav1dFilterMode;
+pub const DAV1D_FILTER_8TAP_SMOOTH: Dav1dFilterMode =
+    Rav1dFilterMode::Smooth8Tap as Dav1dFilterMode;
+pub const DAV1D_FILTER_8TAP_REGULAR: Dav1dFilterMode =
+    Rav1dFilterMode::Regular8Tap as Dav1dFilterMode;
+
+#[repr(u8)]
```

Comment:
Is `#[repr(u8)]` needed for the `as u8` casts?  Otherwise, we should leave it out, as it prevents niche optimization I believe.